config/README.md

```diff
@@ -0,0 +1,76 @@
+# Configuration
+
+Foundry's configuration system allows you to configure it's tools the way _you_ want while also providing with a
+sensible set of defaults.
+
+## Profiles
+
+Configurations can be arbitrarily namespaced by [`Profile`]s. Foundry's default [`Config`] is also named `default`, but
+can arbitrarily name and configure profiles as you like and set the `FOUNDRY_PROFILE` environment variable to the
+selected profile's name. This results in foundry's tools (forge) preferring the values in the profile with the named
+that's set in `FOUNDRY_PROFILE`.
+
+## foundry.toml
+
+Foundry's tools search for a `foundry.toml`  or the filename in a `FOUNDRY_CONFIG` environment variable starting at the
+current working directory. If it is not found, the parent directory, its parent directory, and so on are searched until
+the file is found or the root is reached. The typical location for the global `foundry.toml` would be `~/foundry.toml`.
+If the path set in `FOUNDRY_CONFIG` is absolute, no such search takes place and the absolute path is used directly.
+
+In `foundry.toml` you can define multiple profiles, therefore the file is assumed to be _nested_, so each top-level key
+declares a profile and its values the value for the profile. The following is an example of what such a file might look
+like:
+
+```toml
+## defaults for _all_ profiles
+[default]
+src = "src"
+out = "out"
+libs = ["lib"]
+solc-version = "8.0.10"
+eth-rpc-url = "https://mainnet.infura.io"
+
+## set only when the `hardhat` profile is selected
+[hardhat]
```

Comment:
When somebody uses `--hardhat` in the cli, should this profile be automatically turned on?

Reply:
since this is the cli flag, it should override the config?
but the hardhat style paths are set during init in the `[default]` profile